rust/kernel/types.rs

```diff
@@ -181,6 +181,44 @@ impl<T: PointerWrapper + Deref> PointerWrapper for Pin<T> {
     }
 }
 
+/// Implements [`PointerWrapper`] for `{i,u}{8,16,32,64,size}`.
+macro_rules! primitive_pointer_wrapper {
+    ($t:ty) => {
+        impl PointerWrapper for $t {
+            // Pointers that carry values for primitives should not be
+            // borrowed, therefore makes `borrow` always return a
+            // reference to `()`.
+            type Borrowed = UnsafeReference<()>;
+
+            fn into_pointer(self) -> *const c_types::c_void {
+                self as usize as *mut _
+            }
+
+            unsafe fn borrow(_ptr: *const c_types::c_void) -> Self::Borrowed {
+                // SAFETY: `()` always remains valid and immutable.
+                unsafe { UnsafeReference::new(&()) }
+            }
+
+            unsafe fn from_pointer(ptr: *const c_types::c_void) -> $t {
+                ptr as usize as $t
+            }
+        }
+    };
+}
+primitive_pointer_wrapper! {i8}
+primitive_pointer_wrapper! {i16}
+primitive_pointer_wrapper! {i32}
+primitive_pointer_wrapper! {isize}
+primitive_pointer_wrapper! {u8}
+primitive_pointer_wrapper! {u16}
+primitive_pointer_wrapper! {u32}
+primitive_pointer_wrapper! {usize}
+
+#[cfg(target_pointer_width = "64")]
+primitive_pointer_wrapper! {i64}
```

Comment:
Is this really necessary? It might be a bit confusing that `PointerWrapper` is conditionally implemented for `i64` depending on arch. This makes it more difficult to ensure drivers are as pointer-size agnostic as possible.

I think we should just leave out `i64/u64`. If users are 100% sure that the code will only ever be used in 64-bit arch, they can simply cast the number to/from `isize/usize`.

Reply:
I agree with your concern, and currently I cannot think of any case that has to rely on `i64/u64` being a `PointerWrapper`. I only put it here because of completeness.

@wedsonaf Do you have any case in mind that requires `i64/u64` being `PointerWrapper`, if not, I will drop this in the next version.

Reply:
> @wedsonaf Do you have any case in mind that requires `i64/u64` being `PointerWrapper`, if not, I will drop this in the next version.

I don't have anything in mind that requires `i/u64`.